Segmentation fault in xml.go

Hi,

After updating to rules_go version 0.54.0 one test case/suite is generating a segmentation fault post processing the results of the tests. I have not had a chance to trace the fault back to it's origins but this is what I have so far:

Stack backtrace:

```
PASS
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x2 addr=0x0 pc=0x10423e548]

goroutine 1 [running]:
github.com/bazelbuild/rules_go/go/tools/bzltestutil.toXML({0x1043bbc71, 0x27}, 0x14000293b40)
        external/rules_go+/go/tools/bzltestutil/xml.go:191 +0x3e8
github.com/bazelbuild/rules_go/go/tools/bzltestutil.json2xml({0x1044fe958, 0x14000400030}, {0x1043bbc71, 0x27})
        external/rules_go+/go/tools/bzltestutil/xml.go:165 +0xd94
github.com/bazelbuild/rules_go/go/tools/bzltestutil.writeReport({{0x1400041a000, 0x6d93, 0x8000}, 0x0, 0x0}, {0x1043bbc71, 0x27}, {0x14000018310, 0xad})
        external/rules_go+/go/tools/bzltestutil/wrap.go:167 +0xa0
github.com/bazelbuild/rules_go/go/tools/bzltestutil.Wrap({0x1043bbc71, 0x27})
        external/rules_go+/go/tools/bzltestutil/wrap.go:155 +0x888
main.main()
        bazel-out/darwin_arm64-fastbuild/bin/pkg/adr/adr_test_/testmain.go:97 +0x38
================================================================================
```

Line 91 dereferencing the case duration value is generating the fault.

```go
if name == suiteName {
			duration := *c.duration
		
```

--- a/go/tools/bzltestutil/xml.go
+++ b/go/tools/bzltestutil/xml.go
@@ -188,7 +188,10 @@ func toXML(pkgName string, testcases map[string]*testCase) *xmlTestSuites {
 		}
 		c := testcases[name]
 		if name == suiteName {
-			duration := *c.duration
+			var duration float64
+			if c.duration != nil {
+				duration = *c.duration
+			}
 			if c.start != nil && c.end != nil {
 				// the duration of a test suite may be greater than c.duration
 				// when any test case uses t.Parallel().

